command/agent/template/template.go

```diff
@@ -142,6 +151,14 @@ func (ts *Server) Run(ctx context.Context, incoming chan string, templates []*ct
 						Token: latestToken,
 					},
 				}
+
+				// If we're testing, limit retries to 3 attempts to avoid
```

Comment:
Instead of putting test-only logic, what about specifying something like `retryAttempts` and then doing a conditional off of that?

```go
if ts.retryAttempts >= 0 {
  ctv.Vault.Retry = &ctconfig.RetryConfig{Attempts: &ts.retryAttempts}
}
```

That way it's more flexible and isn't a test-only path in production code.

Reply:
Yep, I made this change earlier but forgot to push.